spec/models/job_proxy_dispatcher_spec.rb

```diff
@@ -172,5 +175,96 @@ module JobProxyDispatcherSpec
         end
       end
     end
+
+    context "with container and vms jobs" do
+      before(:each) do
+        @jobs = (@vms + @repo_vms + @container_images).collect(&:scan)
+        @dispatcher = JobProxyDispatcher.new
+      end
+
+      describe "#pending_jobs" do
+        it "returns only vm jobs by default" do
+          jobs = @dispatcher.pending_jobs
+          expect(jobs.count).to eq(@vms.count + @repo_vms.count)
+          expect(jobs.all? { |x| x.target_class == 'VmOrTemplate' }).to eq(true)
+          expect(jobs.count > 0).to be_truthy # in case something unexpected goes wrong
+        end
+
+        it "returns only container images jobs when requested" do
+          jobs = @dispatcher.pending_jobs(ContainerImage)
+          expect(jobs.count).to eq(@container_images.count)
+          expect(jobs.all? { |x| x.target_class == 'ContainerImage' }).to eq(true)
```

Comment:
does this provide a good error message when this fails?  I'm afraid we'll get something like "false is not true".

Maybe do an assertion for each?

```ruby
jobs.each do |x| 
  expect(x.target_class).to eq 'ContainerImage'
end
```